src/S3/Result/AwsObject.php

```diff
@@ -70,7 +70,7 @@ public function getKey(): ?string
         return $this->Key;
     }
 
-    public function getLastModified(): ?\DateTimeImmutable
+    public function getLastModified(): ?\DateTimeInterface
```

Comment:
Sometime we typehint `DateTimeImmutable` and sometime with `DateTimeInterface` I choosed to always typehint with `DateTimeInterface`